flow/directives.py

```diff
@@ -377,7 +381,7 @@ def _is_fraction(value):
 walltime. Defaults to 12 hours.
 """
 
-_MEMORY = _Directive("memory", validator=_positive_real, default=4)
+_MEMORY = _Directive("memory", validator=_positive_real, default=None)
```

Comment:
Is there a reason not to support units other than Gb? I see that the function for parsing of string memory flags is marked for deprecation because we'll be removing the `--memory` CLI arguments, but we could still support passing a string here. I have a slight preference for supporting this since it's a more natural mapping to the interface that users are accustomed to when manually writing cluster submission scripts, and I don't think there's much overhead to doing so (we can still default to Gb if no units are provided).

Reply:
This does make the code slightly more complex since the templates for different schedulers can't assume a suffix. I'm fine skipping this for now if you can't find a clean and simple solution, we can always add it later if people want that support.

Reply:
We now support

1. Positive numeric value with suffix "g" or "G" indicating memory requested in gigabytes.
2. Positive numeric value with suffix "m" or "M" indicating memory requested in megabytes.
3. Positive numeric value with no suffix indicating memory requested in gigabytes.